src/class-wp-import.php

```diff
@@ -521,9 +521,9 @@ public function get_author_mapping() {
 	 * Doesn't create a new category if its slug already exists.
 	 */
 	public function process_categories() {
-		$this->categories = (array) apply_filters( 'wp_import_categories', $this->categories );
+		$this->categories = apply_filters( 'wp_import_categories', $this->categories );
 
-		if ( empty( $this->categories ) ) {
+		if ( ! is_array( $this->categories ) || empty( $this->categories ) ) {
```

Comment:
1. My previous remark was that this kind of logic change doesn't belong in this PR. Changing it from a cast to an `is_array()` check, doesn't address that.
2. The `is_array()` and `empty()` need to be reversed as you are now testing the variable for being an array before you've tested whether it is actually set. In the above, that's not an issue, but in at least one of the other places it is nd it's the reason the unit tests are failing for this PR.
```php
if ( empty( $this->categories ) || ! is_array( $this->categories ) ) {
```